commit d92f7f0232a16c345aee07702c9e6700d01a94cb
Author: malahal naineni <malahal@us.ibm.com>
Date:   Thu Jul 21 20:35:06 2011 +0000

    nfs4-acl-tools: Fix segfault if format of the input file is incorrect
    
    The sizeof operator, when applied to a parameter declared to have array,
    yields the size of the adjusted (pointer) type, even if the parameter
    declaration specifies a length.
    
    Signed-off-by: Malahal Naineni <malahal@us.ibm.com>
    Signed-off-by: J. Bruce Fields <bfields@redhat.com>

diff --git a/libnfs4acl/nfs4_ace_from_string.c b/libnfs4acl/nfs4_ace_from_string.c
index 9d877fb..462fcc0 100644
--- a/libnfs4acl/nfs4_ace_from_string.c
+++ b/libnfs4acl/nfs4_ace_from_string.c
@@ -100,7 +100,7 @@ parse_alloc_fields(char *buf, char *fields[NUMFIELDS])
 	if (!buf)
 		return -EINVAL;
 
-	memset(fields, 0, sizeof(fields));
+	memset(fields, 0, sizeof(char *) * NUMFIELDS);
 
 	for (i = 0; buf[i] != '\0'; i++) {
 		if (buf[i] == ':')
